README.md
# spring17-devops
DevOps packages for deploying scalable microservice architectural components for Spring 2017 class excercises


history.txt
    1  clear
    2  pwd
    3  wget https://raw.githubusercontent.com/airavata-courses/TeamOmni/8db4870ac17b5d4a7aae17699fa5ab3ac589fafc/scripts/requirements/DockerInstall.sh
    4  ls -lrt
    5  chmod 777
    6  chmod 777 DockerInstall.sh 
    7  ls -lrt
    8  ./DockerInstall.sh
    9  ls -lrt
   10  docker man
   11  sudo docker
   12  clear
   13  sudo docker ps -a
   14  clear
   15  ls -lrt
   16  history
   17  docker run -d -v /var/run/docker.sock:/tmp/docker.sock --name registrator -h registrator gliderlabs/registrator:latest consul://127.0.0.1:8500
   18  sudo docker run -d -v /var/run/docker.sock:/tmp/docker.sock --name registrator -h registrator gliderlabs/registrator:latest consul://127.0.0.1:8500
   19  clear
   20  sudo docker ps -a
   21  clear
   22  sudo docker ps -a
   23  sudo docker run -d --name=registrator --net=host --volume=/var/run/docker.sock:/tmp/docker.sock gliderlabs/registrator:latest consul://localhost:8500
   24  docker rm -f b0c0b7c57cd4
   25  sudo docker rm -f b0c0b7c57cd4
   26  sudo docker run -d --name=registrator --net=host --volume=/var/run/docker.sock:/tmp/docker.sock gliderlabs/registrator:latest consul://localhost:8500
   27  sudo docker ps -a
   28  sudo docker run -d -P --name node1 -h node1 jlordiales/python-micro-service:latest
   29  sudo docker ps -a
   30  curl
   31  sudo apt-get install apt-transport-https                        ca-certificates
   32  clear
   33  curl -fsSL https://yum.dockerproject.org/gpg | sudo apt-key add -
   34  apt-key fingerprint 58118E89F3A912897C070ADBF76221572C52609D
   35  sudo add-apt-repository        "deb https://apt.dockerproject.org/repo/ \
   36         ubuntu-$(lsb_release -cs) \
   37         main"
   38  sudo apt-get update
   39  sudo apt-get -y install docker-engine
   40  clear
   41  sudo docker run hello-world
   42  ls -lrt
   43  clear
   44  history
   45  docker run -p 8400:8400 -p 8500:8500 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise $DOCKER_IP -bootstrap
   46  sudo docker run -p 8400:8400 -p 8500:8500 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise $DOCKER_IP -bootstrap
   47  clear
   48  docker ps -a
   49  sudo docker ps -a
   50  clear
   51  sudo docker ps -a
   52  docker rm -f consul
   53  sudo docker rm -f consul
   54  sudo docker rm -f cocky_turing
   55  clear
   56  sudo docker ps -a
   57  sudo docker run -p 8400:8400 -p 8500:8500 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise $DOCKER_IP -bootstrap
   58  boot2docker ip
   59  clear
   60  docker ps -a
   61  sudo docker ps -a
   62  sudo docker rm -f consul
   63  sudo docker run -p 8400:8400 -p 8500:8500 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise 127.0.0.1 -bootstrap
   64  clear
   65  docker ps -a
   66  sudo docker ps -a
   67  docker rm -f 9c9d9a0b6ba8
   68  sudo docker rm -f 9c9d9a0b6ba8
   69  clear
   70  sudo docker ps -a
   71  sudo docker kill db96c36c30d6
   72  sudo docker kill e368eeeacd9b
   73  sudo docker ps -a
   74  sudo docker rm -f db96c36c30d6
   75  sudo docker rm -f e368eeeacd9b
   76  clear
   77  sudo docker run -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap
   78  docker ps -a 
   79  sudo docker ps -a
   80  clear
   81  sudo docker ps -a
   82  sudo docker run -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
   83  touch /var/log/sga-consul.log
   84  sudo touch /var/log/sga-consul.log
   85  sudo chmod 777 /var/log/sga-consul.log
   86  clear
   87  sudo docker run -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
   88  tail -f /var/log/sga-consul.log
   89  clear
   90  sudo docker ps -a
   91  sudo docker rm -f 66287e65f702
   92  sudo docker run -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
   93  tail -f /var/log/sga-consul.log
   94  sudo docker run -d -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
   95  tail -f /var/log/sga-consul.log
   96  clear
   97  docker ps -a
   98  sudo docker ps -a
   99  sudo docker rm -f 59f0ba61d990
  100  clear
  101  sudo docker ps -a
  102  sudo docker run -d -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
  103  tail -f /var/log/sga-consul.log
  104  clear
  105  sudo docker ps -a
  106  history
  107  clear
  108  sudo docker ps -a
  109  sudo docker rm -f ffc8b40accca
  110  sudo docker run -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
  111  sudo docker run -p 8400:8400 -p 8500:8500 -p 8301:8301 -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap
  112  sudo docker ps -a
  113  sudo docker rm -f 53561d971757
  114  free -m

DockerCommands.sh
##Install Docker on ubuntu
 sudo apt-get install apt-transport-https                        ca-certificates
 curl -fsSL https://yum.dockerproject.org/gpg | sudo apt-key add -
 apt-key fingerprint 58118E89F3A912897C070ADBF76221572C52609D
 sudo add-apt-repository        "deb https://apt.dockerproject.org/repo/ \
       ubuntu-$(lsb_release -cs) \
       main"
 sudo apt-get update
 sudo apt-get -y install docker-engine
 
 
 ##Starts consul server inside docker ubuntu
 sudo docker run -d -p 8400:8400 -p 8500:8500 -p 8300:8300 -p 8301:8301 -p 8301:8301/udp -p 8302:8302 -p 8302:8302/udp -p 8600:53/udp -h consul --name consul progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -bootstrap >> /var/log/sga-consul.log 2>&1
 ## UI - IP:8500/ui
 
## Start consul agent on laravel instance
sudo docker run -d -p 8400:8400 -p 8500:8500 -p 8300:8300 -p 8301:8301 -p 8301:8301/udp -p 8302:8302 -p 8302:8302/udp -p 8600:53/udp -h consulAgent1 --name consulAgent1 progrium/consul -server -advertise `wget -qO- http://instance-data/latest/meta-data/public-ipv4` -join 52.14.33.15 >> /var/log/sga-consul.log 2>&1
 
 
 ## Start registrator 
 sudo docker run -d --name=registrator --net=host --volume=/var/run/docker.sock:/tmp/docker.sock gliderlabs/registrator:latest consul://localhost:8500
 
 
 sudo docker build --no-cache -t sga/laravel:v1 .
 
 sudo docker run -it --name laravelApp -v /home/ubuntu/laravel-portal/spring17-laravel-portal/:/var/www/laravel/ -h sga.portal -p 8089:80 -d sga/laravel:v1
